File: JavaPrototypes/ExpenseCalculator/src/com/example/expenseCalculator/EmployeeFactory.java
package com.example.expenseCalculator;

public class EmployeeFactory {

	public static Employee create(EmployeeType employeeType, String title, Double expenseAllocation) {
		switch (employeeType) {
			case Manager:
				return new Manager(title, expenseAllocation);
			default:
				return new EmployeeInst(title, expenseAllocation);
		}
	}
}

File: DotNetPrototypes/ExpenseCalculator/EmployeeFactory.cs
namespace ExpenseCalculator
{
    public class EmployeeFactory
    {
        public static IEmployee Create(EmployeeType employeeType, string title, decimal expenseAllocation)
        {
            switch (employeeType) { 
                case EmployeeType.Manager:
                    return new Manager(title, expenseAllocation);
                default:
                    return new Employee(title, expenseAllocation);
            }
        }
    }

    public enum EmployeeType { 
        Manager,
        Employee
    }
}
